gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/rollup/RollUpTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.rollup;
+
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.api.epgm.GraphCollection;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.operators.combination.ReduceCombination;
+import org.gradoop.flink.model.impl.operators.grouping.Grouping;
+import org.gradoop.flink.model.impl.operators.grouping.functions.aggregation.CountAggregator;
+import org.gradoop.flink.model.impl.operators.grouping.functions.aggregation.PropertyValueAggregator;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A test for {@link RollUp}, calling the operator and checking if the result is
+ * correct.
+ */
+public class RollUpTest extends GradoopFlinkTestBase {
+
+  /**
+   * Executes a rollUp on vertices using a single grouping key and checks if the result
+   * is correct.
+   *
+   * @throws Exception If the execution fails.
+   */
+  @Test
+  public void testVertexRollUpWithSingleGroupingKey() throws Exception {
+    FlinkAsciiGraphLoader loader = getSocialNetworkLoader();
+
+    LogicalGraph input = loader.getGraphCollectionByVariables("g0", "g1", "g2")
+      .reduce(new ReduceCombination());
+
+    //expected
+    loader.initDatabaseFromString("g0:g0 {vertexRollUpGroupingKeys:\"age\"}[" +
+      "(v__0 {count:1L,age:20})" +
+      "(v__1 {count:2L,age:30})" +
+      "(v__2 {count:2L,age:35})" +
+      "(v__3 {count:1L,age:40})" +
+      "(v__2)-[e__0]->(v__0)" +
+      "(v__1)-[e__1]->(v__0)" +
+      "(v__3)-[e__2]->(v__1)" +
+      "(v__0)-[e__3]->(v__1)" +
+      "(v__2)-[e__4]->(v__1)" +
+      "(v__2)-[e__5]->(v__3)" +
+      "(v__1)-[e__6]->(v__1)" +
+      "(v__1)-[e__7]->(v__3)" +
+      "]");
+    GraphCollection expected = loader.getGraphCollection();
+
+    List<String> vertexGK = new ArrayList<>(Arrays.asList("age"));
```

Comment:
Use `Collections.singletonList()` 
```suggestion
    List<String> vertexGK = Collections.singletonList("age");
```

Reply:
Only works for lists not used for building the grouping key combinations as the internal function requires concrete implementations and must not recieve an abstract list.

Reply:
OK then refactor your combination function. Since Abstract lists are supported by the groupBy operator, we will have the same support here.
In the `createGroupingKeyCombinations` function, 
```
...
    // put the list in an ArrayList instance to support removing items
    List<String> tempGroupingKeys = new ArrayList<>(groupingKeys);
    while (!tempGroupingKeys.isEmpty()) {
      combinations.add(new ArrayList<>(tempGroupingKeys));
      tempGroupingKeys.remove(tempGroupingKeys.size() - 1);
    }
   ...
```